src/mips/common/util/gen-sjis-encode-table.py

```
@@ -0,0 +1,51 @@
+# We walk every forward entry, reconstruct its Shift-JIS bytes, and keep the
+# lowest Shift-JIS code per Unicode value (so single-byte forms win over
+# double-byte). Run from the repo root.
+import re, sys
+
+fwd = [int(x, 16) for x in
+       re.findall(r'0x[0-9a-fA-F]{4}', open('src/mips/common/util/sjis-table.h').read())]
+
+def index_to_sjis(index):
+    if index < 0x100:
+        return index
+    if index < 0x1100:   base, lead_hi = 0x100,  0x80
```

Ruff (0.15.18)

[error] Multiple statements on one line (colon)

(E701)